18_reinforcement_learning.ipynb

```diff
@@ -1925,12 +1924,12 @@
     "        best_weights = model.get_weights()\n",
     "        best_score = step\n",
     "    print(\"\\rEpisode: {}, Steps: {}, eps: {:.3f}\".format(episode, step + 1, epsilon), end=\"\")\n",
-    "    if episode > 50:\n",
-    "        training_step(batch_size)\n",
     "    if episode % 50 == 0:\n",
     "        target.set_weights(model.get_weights())\n",
+    "    if episode >= 50:\n",
```

Comment:
I'm not sure why `set_weights()` needs to happen before the `training_set()`? Could you please explain this?

Reply:
Suppose`episode = 50`. Then (as per my comment on line 1760) we will call `training_step(batch_size)` for the first time. But first, we have to call `target.set_weights(model.get_weights())`.

Reply:
Ooooh, I get it now, thanks for the explanation. 🙏  Yes, this would work fine.

Okay, so if the goal is to avoid the redundant call to `set_weights()` at the very beginning of training, I think I'd prefer a smaller change, replacing:

```python
    if episode >= 50:
        training_step(batch_size)
    if episode % 50 == 0:
        target.set_weights(model.get_weights())
```

with:

```python
    if episode >= 50:
        training_step(batch_size)
        if episode % 50 == 0:
            target.set_weights(model.get_weights())
```

I've just indented the last 2 lines. In Python can you fix a program by just adding a few spaces! 😆 

Other than changing slightly less code, I think it's safer to keep `clone_model()` and `set_weights()` next to each other when the intention is to clone both a model's architecture and its weights. Or else we have model with the right architecture but invalid weights, and it's easy to use the invalid model by mistake, before it is correctly initialized.

Of course your solution avoids this issue, but there's still a risk. For example, if someone wants to try soft updates, they will presumably comment out the current code and uncomment the soft updates section like this:

```python
    #if episode % 50 == 0:
    #    target.set_weights(model.get_weights())
    #if episode >= 50:
    #    training_step(batch_size)
    # Alternatively, you can do soft updates at each step:
    if episode >= 50:
        target_weights = target.get_weights()
        online_weights = model.get_weights()
        for index in range(len(target_weights)):
            target_weights[index] = 0.99 * target_weights[index] + 0.01 * online_weights[index]
        target.set_weights(target_weights)
```

Note that we're now using `target.get_weights()` before it's ever properly initialized.

If you agree, then could you please update the PR to keep just the `>=50` changes? If it's too much hassle, don't worry, I can do it.

Reply:
Sorry, I meant keep just the `>=50` changes & the indentation changes.